<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Part 1. Examining the Data | BioDIGS: Exploring Soil Data</title>
  <meta name="description" content="This activity guides students through exploration of the BioDIGS soil data using the tidyverse in R." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Part 1. Examining the Data | BioDIGS: Exploring Soil Data" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This activity guides students through exploration of the BioDIGS soil data using the tidyverse in R." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Part 1. Examining the Data | BioDIGS: Exploring Soil Data" />
  
  <meta name="twitter:description" content="This activity guides students through exploration of the BioDIGS soil data using the tidyverse in R." />
  



<meta name="date" content="2025-02-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/GDSCN_style/gdscn_favicon.ico" type="image/x-icon" />
<link rel="prev" href="introduction.html"/>
<link rel="next" href="part-2.-summarizing-with-statistics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
<link rel="stylesheet" href="assets/GDSCN_style/gdscn.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://www.gdscn.org/" target="_blank"><img src="assets/GDSCN_style/logo-gdscn.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#target-audience"><i class="fa fa-check"></i><b>0.1</b> Target Audience</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#platform"><i class="fa fa-check"></i><b>0.2</b> Platform</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>0.3</b> Data</a></li>
</ul></li>
<li class="part"><span><b>BioDIGS Overview</b></span></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Background</a>
<ul>
<li class="chapter" data-level="1.1" data-path="background.html"><a href="background.html#what-is-genomics"><i class="fa fa-check"></i><b>1.1</b> What is genomics?</a></li>
<li class="chapter" data-level="1.2" data-path="background.html"><a href="background.html#what-is-data-science"><i class="fa fa-check"></i><b>1.2</b> What is data science?</a></li>
<li class="chapter" data-level="1.3" data-path="background.html"><a href="background.html#what-is-cloud-computing"><i class="fa fa-check"></i><b>1.3</b> What is cloud computing?</a></li>
<li class="chapter" data-level="1.4" data-path="background.html"><a href="background.html#why-soil-microbes"><i class="fa fa-check"></i><b>1.4</b> Why soil microbes?</a></li>
<li class="chapter" data-level="1.5" data-path="background.html"><a href="background.html#heavy-metals-and-human-health"><i class="fa fa-check"></i><b>1.5</b> Heavy metals and human health</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="research-team.html"><a href="research-team.html"><i class="fa fa-check"></i><b>2</b> Research Team</a>
<ul>
<li class="chapter" data-level="2.1" data-path="research-team.html"><a href="research-team.html#soil-sampling"><i class="fa fa-check"></i><b>2.1</b> Soil sampling</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="support.html"><a href="support.html"><i class="fa fa-check"></i><b>3</b> Support</a>
<ul>
<li class="chapter" data-level="3.1" data-path="support.html"><a href="support.html#funding"><i class="fa fa-check"></i><b>3.1</b> Funding</a></li>
<li class="chapter" data-level="3.2" data-path="support.html"><a href="support.html#sponsors"><i class="fa fa-check"></i><b>3.2</b> Sponsors</a></li>
<li class="chapter" data-level="3.3" data-path="support.html"><a href="support.html#analytical-and-computational-support"><i class="fa fa-check"></i><b>3.3</b> Analytical and Computational Support</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="biodigs-data.html"><a href="biodigs-data.html"><i class="fa fa-check"></i><b>4</b> BioDIGS Data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="biodigs-data.html"><a href="biodigs-data.html#sample-metadata"><i class="fa fa-check"></i><b>4.1</b> Sample Metadata</a></li>
<li class="chapter" data-level="4.2" data-path="biodigs-data.html"><a href="biodigs-data.html#soil-property-data"><i class="fa fa-check"></i><b>4.2</b> Soil Property Data</a></li>
<li class="chapter" data-level="4.3" data-path="biodigs-data.html"><a href="biodigs-data.html#genomics-and-metagenomics-data-and-metadata"><i class="fa fa-check"></i><b>4.3</b> Genomics and Metagenomics Data and Metadata</a></li>
<li class="chapter" data-level="4.4" data-path="biodigs-data.html"><a href="biodigs-data.html#biodigsdata-r-package"><i class="fa fa-check"></i><b>4.4</b> BioDIGSData R package</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="biodigs-data.html"><a href="biodigs-data.html#installation"><i class="fa fa-check"></i><b>4.4.1</b> Installation <img src="resources/images/hex_logo.png" align="right" height="138" /></a></li>
<li class="chapter" data-level="4.4.2" data-path="biodigs-data.html"><a href="biodigs-data.html#usage"><i class="fa fa-check"></i><b>4.4.2</b> Usage</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>AnVIL Overview</b></span></li>
<li class="chapter" data-level="5" data-path="billing.html"><a href="billing.html"><i class="fa fa-check"></i><b>5</b> Billing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="billing.html"><a href="billing.html#create-google-billing-account"><i class="fa fa-check"></i><b>5.1</b> Create Google Billing Account</a></li>
<li class="chapter" data-level="5.2" data-path="billing.html"><a href="billing.html#add-terra-to-google-billing-account"><i class="fa fa-check"></i><b>5.2</b> Add Terra to Google Billing Account</a></li>
<li class="chapter" data-level="5.3" data-path="billing.html"><a href="billing.html#add-members-to-google-billing-account"><i class="fa fa-check"></i><b>5.3</b> Add Members to Google Billing Account</a></li>
<li class="chapter" data-level="5.4" data-path="billing.html"><a href="billing.html#set-alerts-for-google-billing"><i class="fa fa-check"></i><b>5.4</b> Set Alerts for Google Billing</a></li>
<li class="chapter" data-level="5.5" data-path="billing.html"><a href="billing.html#view-spend-for-google-billing"><i class="fa fa-check"></i><b>5.5</b> View Spend for Google Billing</a></li>
<li class="chapter" data-level="5.6" data-path="billing.html"><a href="billing.html#create-terra-billing-project"><i class="fa fa-check"></i><b>5.6</b> Create Terra Billing Project</a></li>
<li class="chapter" data-level="5.7" data-path="billing.html"><a href="billing.html#add-member-to-terra-billing-project"><i class="fa fa-check"></i><b>5.7</b> Add Member to Terra Billing Project</a></li>
<li class="chapter" data-level="5.8" data-path="billing.html"><a href="billing.html#disable-terra-billing-project"><i class="fa fa-check"></i><b>5.8</b> Disable Terra Billing Project</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="anvil-workspace.html"><a href="anvil-workspace.html"><i class="fa fa-check"></i><b>6</b> AnVIL Workspace</a>
<ul>
<li class="chapter" data-level="6.1" data-path="anvil-workspace.html"><a href="anvil-workspace.html#create-google-account"><i class="fa fa-check"></i><b>6.1</b> Create Google Account</a></li>
<li class="chapter" data-level="6.2" data-path="anvil-workspace.html"><a href="anvil-workspace.html#create-workspace"><i class="fa fa-check"></i><b>6.2</b> Create Workspace</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="programming-platforms.html"><a href="programming-platforms.html"><i class="fa fa-check"></i><b>7</b> Programming Platforms</a>
<ul>
<li class="chapter" data-level="7.1" data-path="programming-platforms.html"><a href="programming-platforms.html#video-overview-on-using-rstudio"><i class="fa fa-check"></i><b>7.1</b> Video overview on using RStudio</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="programming-platforms.html"><a href="programming-platforms.html#objectives"><i class="fa fa-check"></i><b>7.1.1</b> Objectives</a></li>
<li class="chapter" data-level="7.1.2" data-path="programming-platforms.html"><a href="programming-platforms.html#slides"><i class="fa fa-check"></i><b>7.1.2</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="programming-platforms.html"><a href="programming-platforms.html#launching-rstudio"><i class="fa fa-check"></i><b>7.2</b> Launching RStudio</a></li>
<li class="chapter" data-level="7.3" data-path="programming-platforms.html"><a href="programming-platforms.html#touring-rstudio"><i class="fa fa-check"></i><b>7.3</b> Touring RStudio</a></li>
<li class="chapter" data-level="7.4" data-path="programming-platforms.html"><a href="programming-platforms.html#pausing-rstudio"><i class="fa fa-check"></i><b>7.4</b> Pausing RStudio</a></li>
<li class="chapter" data-level="7.5" data-path="programming-platforms.html"><a href="programming-platforms.html#deleting-rstudio"><i class="fa fa-check"></i><b>7.5</b> Deleting RStudio</a></li>
</ul></li>
<li class="part"><span><b>Student Activity</b></span></li>
<li class="chapter" data-level="8" data-path="download.html"><a href="download.html"><i class="fa fa-check"></i><b>8</b> Download</a></li>
<li class="chapter" data-level="9" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>9</b> Introduction</a>
<ul>
<li class="chapter" data-level="9.1" data-path="introduction.html"><a href="introduction.html#before-you-start"><i class="fa fa-check"></i><b>9.1</b> Before You Start</a></li>
<li class="chapter" data-level="9.2" data-path="introduction.html"><a href="introduction.html#objectives-1"><i class="fa fa-check"></i><b>9.2</b> Objectives</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="part-1.-examining-the-data.html"><a href="part-1.-examining-the-data.html"><i class="fa fa-check"></i><b>10</b> Part 1. Examining the Data</a></li>
<li class="chapter" data-level="11" data-path="part-2.-summarizing-with-statistics.html"><a href="part-2.-summarizing-with-statistics.html"><i class="fa fa-check"></i><b>11</b> Part 2. Summarizing with Statistics</a></li>
<li class="chapter" data-level="12" data-path="part-3.-visualizing-the-data.html"><a href="part-3.-visualizing-the-data.html"><i class="fa fa-check"></i><b>12</b> Part 3. Visualizing the Data</a></li>
<li class="chapter" data-level="13" data-path="activity-questions.html"><a href="activity-questions.html"><i class="fa fa-check"></i><b>13</b> Activity Questions</a>
<ul>
<li class="chapter" data-level="13.1" data-path="activity-questions.html"><a href="activity-questions.html#part-1.-examining-the-data-1"><i class="fa fa-check"></i><b>13.1</b> Part 1. Examining the Data</a></li>
<li class="chapter" data-level="13.2" data-path="activity-questions.html"><a href="activity-questions.html#part-2.-summarizing-the-data-with-statistics"><i class="fa fa-check"></i><b>13.2</b> Part 2. Summarizing the Data with Statistics</a></li>
<li class="chapter" data-level="13.3" data-path="activity-questions.html"><a href="activity-questions.html#part-3.-visualizing-the-data-1"><i class="fa fa-check"></i><b>13.3</b> Part 3. Visualizing the Data</a></li>
</ul></li>
<li class="part"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="https://hutchdatascience.org/"> The Fred Hutch Data Science Lab </a></p>
<a href="https://hutchdatascience.org/" target="_blank"><img src="https://raw.githubusercontent.com/jhudsl/OTTR_Template/main/style-sets/fhdasl/copy_to_assets/big-dasl-stacked.png" style="width: 80%; padding-left: 34px; padding-top: 8px;"</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
<p style="padding-left: 40px;"><div class="trapezoid" style = "padding-left: 40px;"><span><a href="https://docs.google.com/forms/d/e/1FAIpQLScrDVb_utm55pmb_SHx-RgELTEbCCWdLea0T3IzS0Oj00GE4w/viewform?usp=pp_url&entry.1565230805=BioDIGS:+Exploring+Soil+Data"> Click here to provide feedback</a> <img src="assets/itcr_arrow.png" style=" width: 10%" ></span></div></p>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BioDIGS: Exploring Soil Data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
</head>



<div class="hero-image-container">
  <img class= "hero-image" src="assets/GDSCN_style/dasl_thin_main_image.png">
</div>
<div id="part-1.-examining-the-data" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Part 1. Examining the Data<a href="part-1.-examining-the-data.html#part-1.-examining-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>We will use the <code>BioDIGSData</code> package to retrieve the data. We first need to install the package from where it is stored on GitHub.</p>
<div class="dictionary">
<p><strong>Packages and libraries</strong></p>
<p>You might see or hear the term “package” or “library” when working with the R programming language.</p>
<p>Packages are collections of R functions, data, and documentation that extend the base functionality of R. They are the fundamental units of shareable code in R. Packages are developed by the R community and made available through repositories like CRAN (Comprehensive R Archive Network), Bioconductor, and GitHub. When you install a package, you gain access to all the functions, data, and documentation provided by that package.</p>
<p>Libraries are the directories where packages are stored.</p>
<p>We also use the <code>library</code> command to load and attach packages to the R environment. When you load a package using <code>library(package_name)</code>, you make the functions and objects from a package available for use in your current R session.</p>
</div>
<p>You may first need to install the <code>remotes</code> package to your RStudio environment. <code>remotes</code> is a package (or chunk of pre-written code) that allows you to download code that has been stored on GitHub into RStudio.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="part-1.-examining-the-data.html#cb5-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;remotes&quot;</span>)</span>
<span id="cb5-2"><a href="part-1.-examining-the-data.html#cb5-2" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;fhdsl/BioDIGSData&quot;</span>, <span class="at">upgrade =</span> <span class="st">&quot;never&quot;</span>)</span></code></pre></div>
<p>If you’re having trouble with the code above, you can also try:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="part-1.-examining-the-data.html#cb6-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;remotes&quot;</span>)</span>
<span id="cb6-2"><a href="part-1.-examining-the-data.html#cb6-2" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_url</span>(<span class="st">&#39;https://github.com/fhdsl/BioDIGSData/archive/refs/tags/v1.0.0.0.tar.gz&#39;</span>)</span></code></pre></div>
<p>Once you’ve installed the package, we can load the package (which just means we have access to all the data stored in the BioDIGSData package). Then we assign the soil testing data to an <em>object</em>. This command follows the code structure:</p>
<p>dataset_object_name &lt;- stored_BioDIGS_dataset</p>
<p>The “dataset_object_name” is what RStudio will call the dataset after you open it. The “stored_BioDIGS_dataset” is what the dataset is called within the BioDIGSData package. Finally, the arrow (“&lt;-”) tells R to open the “stored_BioDIGS_dataset” and save it in your environment as “dataset_object_name”.</p>
<p>The order of these commands might be odd to our eyes, but it makes perfect sense to RStudio!</p>
<p>The soil testing data is called <code>BioDIGS_soil_data</code> in the BioDIGSData package. When we save this dataset into our environment, we’re calling is <code>soil.values</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="part-1.-examining-the-data.html#cb7-1" tabindex="-1"></a><span class="fu">library</span>(BioDIGSData)</span>
<span id="cb7-2"><a href="part-1.-examining-the-data.html#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="part-1.-examining-the-data.html#cb7-3" tabindex="-1"></a>soil.values <span class="ot">&lt;-</span> <span class="fu">BioDIGS_soil_data</span>()</span></code></pre></div>
<p>It <em>seems</em> like the dataset loaded, but it’s always a good idea to verify. There are many ways to check, but the easiest approach (if you’re using RStudio) is to look at the Environment tab on the upper right-hand side of the screen. You should now have an object called <code>soil.values</code> that includes some number of observations for 28 variables. The <em>observations</em> refer to the number of rows in the dataset, while the <em>variables</em> tell you the number of columns. As long as neither the observations or variables are 0, you can be confident that your dataset loaded.</p>
<p><img src="resources/images/09-soil_exploration_module_files/figure-html/1u2CIcN2AxprMbWLzAldr_V-njdvjCS8HLYjvuy6jmfs_g33497bd5a49_0_9.png" alt="If the dataset loaded, you will see an object with non-zero observations and variables in the Environment tab." width="100%" style="display: block; margin: auto;" /></p>
<p>Let’s take a quick look at the dataset. We can do this by clicking on soil.values object in the Environment tab. (Note: this is equivalent to typing <code>View(soil.values)</code> in the R console.)</p>
<p>This will open a new window for us to scroll through the dataset.</p>
<p><img src="resources/images/09-soil_exploration_module_files/figure-html/1u2CIcN2AxprMbWLzAldr_V-njdvjCS8HLYjvuy6jmfs_g33497bd5a49_0_12.png" alt="You can click on the object in the Environment tab to open a new window that allows you to scroll through the loaded dataset." width="100%" style="display: block; margin: auto;" /></p>
<p>Well, the data definitely loaded, but those column names aren’t immediately understandable. What could <strong>As_EPA3051</strong> possibly mean? In addition to the dataset, we need to load the <em>data dictionary</em> as well.</p>
<div class="dictionary">
<p><strong>Data dictionary:</strong> a file containing the names, definitions, and attributes about data in a database or dataset.</p>
</div>
<p>In this case, the data dictionary can help us make sense of what sort of values each column represents. The data dictionary for the BioDIGS soil testing data is available in the R package (see code below), but we have also reproduced it here.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="part-1.-examining-the-data.html#cb8-1" tabindex="-1"></a>?<span class="fu">BioDIGS_soil_data</span>()</span></code></pre></div>
<p><img src="resources/images/09-soil_exploration_module_files/figure-html/1u2CIcN2AxprMbWLzAldr_V-njdvjCS8HLYjvuy6jmfs_g33497bd5a49_0_15.png" alt="The data dictionary shows up under the Help tab." width="100%" style="display: block; margin: auto;" /></p>
<div class="dictionary">
<ul>
<li><strong>collection_date</strong>: Date sample was collected (soil was removed from a site).</li>
<li><strong>site_id</strong>: Unique letter and number site name. Check BioDIGS_metadata() for GPS coordinates, origin, and more.</li>
<li><strong>sample_id</strong>: Unique sequencing sample identifier.</li>
<li><strong>site_name_rep_detail</strong>: Detailed label for the sample, intended to help disambiguate in case of confusion.</li>
<li><strong>As_EPA3051</strong>: Arsenic (mg/kg), EPA Method 3051A. Quantities &lt; 3.0 are not detectable.</li>
<li><strong>Cd_EPA3051</strong>: Cadmium (mg/kg), EPA Method 3051A. Quantities &lt; 0.2 are not detectable.</li>
<li><strong>Cr_EPA3051</strong>: Chromium (mg/kg), EPA Method 3051A</li>
<li><strong>Cu_EPA3051</strong>: Copper (mg/kg), EPA Method 3051A</li>
<li><strong>Ni_EPA3051</strong>: Nickel (mg/kg), EPA Method 3051A</li>
<li><strong>Pb_EPA3051</strong>: Lead (mg/kg), EPA Method 3051A</li>
<li><strong>Zn_EPA3051</strong>: Zinc (mg/kg), EPA Method 3051A</li>
<li><strong>water_pH</strong>: Water pH</li>
<li><strong>OM_by_LOI_pct</strong>: Organic Matter by Loss on Ignition</li>
<li><strong>P_Mehlich3</strong>: Phosphorus (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>K_Mehlich3</strong>: Potassium (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Ca_Mehlich3</strong>: Calcium (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Mg_Mehlich3</strong>: Magnesium (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Mn_Mehlich3</strong>: Manganese (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Zn_Mehlich3</strong>: Zinc (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Cu_Mehlich3</strong>: Copper (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Fe_Mehlich3</strong>: Iron (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>B_Mehlich3</strong>: Boron (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>S_Mehlich3</strong>: Sulfur (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Na_Mehlich3</strong>: Sodium (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Al_Mehlich3</strong>: Aluminum (mg/kg), using the Mehlich 3 soil test extractant</li>
<li><strong>Est_CEC</strong>: Cation Exchange Capacity (meq/100g) at pH 7.0 (CEC)</li>
<li><strong>Base_Sat_pct</strong>: Base saturation (BS). This represents the percentage of CEC occupied by bases (Ca2+, Mg2+, K+, and Na+). The %BS increases with increasing soil pH (Figure 5). The availability of Ca2+, Mg2+, and K+ increases with increasing %BS.</li>
<li><strong>P_Sat_ratio</strong>: Phosphorus saturation ratio. This is the ratio between the amount of phosphorus present in the soil and the total capacity of that soil to retain phosphorus. The ability of phosphorus to be bound in the soil is primary a function of iron (Fe) and aluminum (Al) content in that soil.</li>
</ul>
</div>
<p>Using the data dictionary, we find that the values in column <code>As_EPA3051</code> give us the arsenic concentration in mg/kg of each soil sample, as determined by <a href="https://www.epa.gov/sites/default/files/2015-12/documents/3051a.pdf">EPA Method 3051A</a>. This method uses a combination of heat and acid to extract specific elements (like arsenic, cadmium, chromium, copper, nickel, lead, and zinc) from soil samples.</p>
<p>While arsenic can occur naturally in soils, higher levels suggest the soil may have been contaminated by mining, hazardous waste, or pesticide application. Arsenic is toxic to humans.</p>
<div class="reflection">
<p>QUESTIONS:</p>
<ol style="list-style-type: decimal">
<li><p>What data is found in the column labeled “Fe_Mehlich3”? Why would we be interested how much of this is in the soil? (You may have to search the internet for this answer.)</p></li>
<li><p>What data is found in the column labeled “Base_Sat_pct”? What does this variable tell us about the soil?</p></li>
</ol>
</div>
<p>You may notice that some cells in the <code>soil.values</code> table contain <em>NA</em>. This just means that the soil testing data for that sample isn’t available yet. We’ll take care of those values in the next part.</p>
<div class="reflection">
<p>QUESTIONS:</p>
<ol start="3" style="list-style-type: decimal">
<li><p>How many observations are in the soil testing values dataset that you loaded? What do each of these observations refer to?</p></li>
<li><p>How many different soil characteristics are in the dataset? How can you tell?</p></li>
</ol>
</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="part-2.-summarizing-with-statistics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
